src/DefaultNodeTypes.jl

```diff
@@ -48,6 +51,32 @@ struct Prior{T} <: IncrementalInference.FunctorSingleton where T <: SamplableBel
 end
 getSample(s::Prior, N::Int=1) = (reshape(rand(s.Z,N),:,N), )
 
+
+# TODO maybe replace X with a type.
+#TODO ParametricTypes or dispatch on each? I like it together for now. Maybe compiler will split it?
+function (s::Prior{<:ParametricTypes})(X1::Vector{Float64};
+                    userdata::Union{Nothing,FactorMetadata}=nothing)
+
+  if isa(s.Z, Normal)
```

Comment:
as above, rather use only MvNormal, rather then battle with API differences and `isa` vs multiple dispatch.